crates/router/src/compatibility/wrap.rs

```diff
@@ -22,7 +22,8 @@ pub async fn compatibility_api_wrap<'a, 'b, A, U, T, Q, F, Fut, S, E>(
 ) -> HttpResponse
 where
     F: Fn(&'b A, U, T) -> Fut,
-    Fut: Future<Output = RouterResult<api::ApplicationResponse<Q>>>,
+    Fut: Future<Output = CustomResult<api::ApplicationResponse<Q>, E2>>,
+    E2: ErrorSwitch<E> + std::error::Error + Send + Sync + 'static,
```

Comment:
do we need the other trait requirements,
```suggestion
    E2: ErrorSwitch<E>,
```

Reply:
@lsampras  any thoughts on this and other suggestions?